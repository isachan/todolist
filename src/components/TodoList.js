import React, { useState, useEffect } from 'react';
import db from '../firebase';
import TodoItem from './TodoItem';

const TodoList = () => {
  const [listData, setListData] = useState([]);
  const [todo, setTodo] = useState('');
  // console.log('rendered!');

  const fetchData = () => {
    db.collection('list').onSnapshot(item =>
      setListData(
        item.docs.map(doc => ({
          id: doc.id, //to trace id
          item: doc.data().item,
          checked: doc.data().checked
        }))
      )
    );
  };

  const handleAdd = async () => {
    await db.collection('list').add({
      item: todo,
      checked: false,
      dateCreated: new Date()
    });
    setTodo('');
    // fetchData();
  };

  const handleCheck = async item => {
    // console.log('item', item);
    await db
      .collection('list')
      .doc(item.id)
      .update({ checked: !item.checked });
  };

  const handleDelete = item => {
    db.collection('list')
      .doc(item.id)
      .delete();
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div className='todo'>
      {/* {console.log('listData', listData)} */}
      <h1 className='todo__title'> Todo list</h1>
      <ul className='todo__list'>
        {listData &&
          listData.map(item => (
            <li key={item.key}>
              <TodoItem
                item={item}
                onCheck={() => handleCheck(item)}
                onDelete={() => handleDelete(item)}
              />
            </li>
          ))}
      </ul>

      <div className='todo__addbar'>
        <input
          className='todo__addbar-input'
          type='text'
          value={todo}
          onChange={e => setTodo(e.target.value)}
        ></input>
        <button
          className='todo__addbar-button'
          id='add-button'
          onClick={handleAdd}
        >
          Add
        </button>
      </div>
    </div>
  );
};

export default TodoList;
