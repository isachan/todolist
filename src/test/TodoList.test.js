import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoList from '../components/TodoList';

describe('TodoList Component', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  it('renders correctly', () => {
    const component = shallow(<TodoList />);
    expect(component).toMatchSnapshot();
  });

  it('should add todo to the list', () => {
    const startTodos = [
      { item: 'mop floor', checked: false, dateCreated: new Date() },
      { item: 'feed cat', checked: false, dateCreated: new Date() }
    ];

    const newTodo = {
      item: 'water plants',
      checked: false,
      dateCreated: new Date()
    };

    const expectedTodos = [
      { item: 'mop floor', checked: false, dateCreated: new Date() },
      { item: 'feed cat', checked: false, dateCreated: new Date() },
      { item: 'water plants', checked: false, dateCreated: new Date() }
    ];

    // need to check how to test for firebase online data
  });
});
