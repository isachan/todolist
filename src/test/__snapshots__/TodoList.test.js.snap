// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`TodoList renders correctly 1`] = `
<div
  className="todo"
>
  <h1
    className="todo__title"
  >
     Todo list
  </h1>
  <div
    className="todo__sort"
  >
    <div
      className="todo__sort-date"
    >
      date
    </div>
    <div
      className="todo__sort-item"
      onClick={[Function]}
    >
      A-Z
    </div>
  </div>
  <ul
    className="todo__list"
  />
  <div
    className="todo__addbar"
  >
    <input
      className="todo__addbar-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="todo__addbar-button"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
